adapters/boldwin_rapid/boldwin_rapid.go

```diff
@@ -0,0 +1,172 @@
+package boldwin_rapid
+
+import (
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/macros"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	//endpoint string
+	endpoint *template.Template
+}
+
+func Builder(_ openrtb_ext.BidderName, config config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	endpointTemplate, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: endpointTemplate,
+	}
+
+	return bidder, nil
+}
+
+func (a adapter) buildEndpointURL(boldwinExt openrtb_ext.ImpExtBoldwinRapid) (string, error) {
+	endpointParams := macros.EndpointTemplateParams{
+		PublisherID: boldwinExt.Pid,
+		PlacementID: boldwinExt.Tid,
+	}
+
+	return macros.ResolveMacros(a.endpoint, endpointParams)
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+
+	for _, imp := range request.Imp {
+		// Create a new request with just this impression
+		reqCopy.Imp = []openrtb2.Imp{imp}
```

Comment:
This doesn't create a new request. Line 48 does. There is only one request copy being created before the loop, so all requests will have the same Imp array with the last Imp. You are likely missing a multi-imp test case.